qiskit/chemistry/results/electronic_structure_result.py

```diff
@@ -67,31 +67,31 @@ def nuclear_dipole_moment(self, value: DipoleTuple) -> None:
     # construct the circuit of the GS from here (if the algorithm supports this)
 
     @property
-    def energy(self) -> Optional[float]:
+    def total_energies(self) -> np.ndarray:
         """ Returns ground state energy if nuclear_repulsion_energy is available from driver """
-        # TODO the fact that this property is computed on the fly breaks the `.combine()`
-        # functionality
-        nre = self.nuclear_repulsion_energy
-        return self.electronic_energy + nre if nre is not None else None
+        nre = self.nuclear_repulsion_energy if self.nuclear_repulsion_energy is not None else 0
+        # Adding float to np.ndarray adds it to each entry
+        return self.electronic_energies + nre
 
     @property
-    def electronic_energy(self) -> float:
+    def electronic_energies(self) -> np.ndarray:
```

Comment:
If this goes over to an array I think it needs more explanation here - having an array of 1 value when computing ground state alone is doable - in the past we had two fields one with just the ground state and one as an array where the lowest index was also the ground state. Its a choice I get it but for a minimum state calculation where you only have one the plural seems odd. Maybe keep electronic_energy property that returns just the contents of first array entry? Similar comment for other fields here

Reply:
because the class is "electronic_structure_result" so it should handle the full electronic structure. actually if one calls the `groundenergy` of the eigenstate result it will return the ground state energy directly.

Reply:
The thing for user of classical drivers is that energy is returned. energy a float no longer exists. I have no issue with energies as an array (we have both the past if you look at the result objects we had) but I think for a user getting ground state calling result.energy is more usable and consistent with what you expected rather than having to do result.energies[0] In fact here the energy property can simply do just that ie return energies[0] so its more natural for a user. I picked on electronic energy here to comment but maybe you could argue it applies less to that as often people want the total.

Reply:
I think it is worst for the structure of this class which is ElectronicStructureResults (and not GroundStateResult) to not work with lists. Not working with list would go reverse to the idea of an electronic **structure**

Reply:
I do not not know what the best way is here - clearly its more cumbersome with lists when the user is just computing a single ground state. 'energy' is normally what classical drivers have as a result field and we followed suit in the past. Either way I guess there is no time for anything more at present.